classes/class-wc-connect-taxjar-integration.php

```diff
@@ -1499,4 +1502,109 @@ public function load_taxjar_admin_new_order_assets() {
 		// Load Javascript for WooCommerce new order page
 		wp_enqueue_script( 'wc-taxjar-order', $this->wc_connect_base_url . 'woocommerce-services-new-order-taxjar-' . WC_Connect_Loader::get_wcs_version() . '.js', array( 'jquery' ), null, true );
 	}
+
+	/**
+	 * Add a 27 cent surcharge to your cart / checkout based on delivery country & state
+	 * Taxes, shipping costs and order subtotal are all included in the surcharge amount
+	 *
+	 * Fee won't be added if all products in the cart are virtual
+	 * 
+	 * Change $fee to set the surcharge to a value to suit
+	 *
+	 * 
+	 * Change $country to set the country
+	 * http://en.wikipedia.org/wiki/ISO_3166-1#Current_codes for available alpha-2 country codes
+	 *
+	 * Change $state to set the state
+	 * Example: The state code for Colorado would be 'CO'
+	 * https://en.wikipedia.org/wiki/ISO_3166-2:US
+	 *
+	 * Uses the WooCommerce fees API
+	 *
+	 * @param WC_Cart $cart WooCommerce Cart object.
+	 */
+	public function add_custom_state_surcharge_fee( $cart ) {
+		if ( is_admin() && ! defined( 'DOING_AJAX' ) ) {
+			return;
+		}
+		
+		if( $this->maybe_all_products_are_virtual( $cart ) ) {
+			return;
+		}
+
+		/** Checks if Local Pickup is selected to not apply the fee */
+		if ( count( array_intersect( wc_get_chosen_shipping_method_ids(), apply_filters( 'woocommerce_local_pickup_methods', array( 'legacy_local_pickup', 'local_pickup' ) ) ) ) > 0 ) {
+			return;
+		}
```

Comment:
We should check if all chosen methods `wc_get_chosen_shipping_method_ids()` are local_pickup, if anyone of them is not fee should be applied.

Reply:
Hi @bartech , I'm not sure if I understand this. Can you explain a bit more on this?

This code is checking if there is any non "local pickup" method is being chosen or not. If it's a local pickup then the fee should not be applied. That's why it has `return` there.

Reply:
As `wc_get_chosen_shipping_method_ids` is an array it can have multiple shipping methods picked (in case there are multiple shipping packages). Now code checks if any method is local pickup and we have to check if all of them are local pickup not just one of them.

Reply:
Ahh ok, I got it now and have made the changes.